core/app/models/spree/stock/simple_coordinator.rb

```diff
@@ -75,7 +75,7 @@ def build_shipments
         # Turn the Stock::Packages into a Shipment with rates
         shipments = packages.map do |package|
           shipment = package.shipment = package.to_shipment
-          shipment.shipping_rates = Spree::Config.stock.estimator_class.new.shipping_rates(package)
+          shipment.shipping_rates = Spree::Config.stock.estimator_class.new(order).shipping_rates(package)
```

Comment:
People may have a custom stock estimator which does not accept the `order` argument. We need to make sure we play nice with that scenario, otherwise this will be a breaking change.

Reply:
The current test coverage provides enough confidence to make sure we're not breaking anything for custom stock estimators, the existing tests still test the case when there is no `order` passed into the constructor and an additional test was added to ensure we use the initializer's `order` if it's passed in https://github.com/solidusio/solidus/blob/7414ae6d2f177610d272130143e92d7e6023dc9d/core/spec/models/spree/stock/estimator_spec.rb#L36

Reply:
@HendrikLouw what I meant is that, if someone provides a custom stock estimator and overrides the `#initialize` method (or if their estimator doesn't inherit from `Spree::Stock::Estimator` at all), their stock estimator's `#initialize` will not accept an `order` argument, but Solidus will still attempt to initialize it with an order argument, which will break the user's code.

Take the following example:

```ruby
# app/models/custom_estimator.rb
class CustomEstimator
  def shipping_rates
    # ...
  end
end

# config/initializers/spree.rb
Spree.config do |config|
  config.stock.estimator_class = 'CustomEstimator'
end

# Somewhere in the code, assuming `order` is a `Spree::Order` instance
Spree::Stock::SimpleCoordinator.new(order).shipments # <= This will raise an error
```

Reply:
Thank you for the full explanation, I've added a way to handle this case https://github.com/solidusio/solidus/pull/4105/files#diff-43c0bdb931bf364be432145396b649919cfee1ce27e82af572bf394f47af8cdaR22, happy with the solution?